### PR TITLE
Create seperate Module for basic git command wrapper

### PR DESCRIPTION
**Pull request by Sinerum**
Create a new module that houses wrapper for basic git commands. And move and refactor the implementations from vara_manager to it and make use of plumbum.

### CODE REVIEW

File: varats-core/varats/utils/git_commands.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+
+def get_branches(
+    repo_folder: Path, extra_args: tp.Optional[tp.List[str]] = None
+) -> str:
+    """Show git branches."""
+    extra_args = [] if extra_args is None else extra_args
+
+    args = ["branch"]
+    args += extra_args
```
Comment (boehmseb) [inline varats-core/varats/utils/git_commands.py]:
```suggestion
    args = ["branch"]
    if extra_args:
        args += extra_args
```

File: varats-core/varats/utils/git_commands.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    return tp.cast(str, git("-C", repo_folder.absolute(), args))
+
+
+def get_tags(repo_folder: Path,
+             extra_args: tp.Optional[tp.List[str]] = None) -> tp.List[str]:
+    """Get the list of available git tags."""
+
+    extra_args = [] if extra_args is None else extra_args
+    args = ["tag"]
+    args += extra_args
```
Comment (boehmseb) [inline varats-core/varats/utils/git_commands.py]:
```suggestion
    args = ["tag"]
    if extra_args:
        args += extra_args
```

File: varats-core/varats/utils/git_commands.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+def fetch_remote(
+    remote: tp.Optional[str] = None,
+    repo_folder: tp.Optional[Path] = None,
+    extra_args: tp.Optional[tp.List[str]] = None
+) -> None:
+    """Fetches the new changes from the remote."""
+    extra_args = [] if extra_args is None else extra_args
+
+    args = ["fetch"]
+    args += extra_args
```
Comment (boehmseb) [inline varats-core/varats/utils/git_commands.py]:
```suggestion
    args = ["fetch"]
    if extra_args:
        args += extra_args
```

**Review by boehmseb — CHANGES_REQUESTED** [review]
Found some instances where code can be simplified.

Also, how did you decide which functions to put in `git_util` and which in `git_commands`?

**Review by boehmseb — APPROVED** [review]

**Review by vulder — APPROVED** [review]
Did you try a "new" blank install of VaRA with vara-buildsetup after moving all the commands?

If not, could you please make a test run :D The install code is not fully tested an I'm a bit scared we could break things there.

**Comment by Sinerum** [comment]
> Did you try a "new" blank install of VaRA with vara-buildsetup after moving all the commands?
> 
> If not, could you please make a test run :D The install code is not fully tested an I'm a bit scared we could break things there.

`vara-buildsetup init` works fine, but building afterwards fails. I am not sure if that is related to this change.

**Review by vulder — APPROVED** [review]

**Pull request merged**